ext/msgpack/unpacker.c

```diff
@@ -343,14 +335,10 @@ static inline int read_raw_body_begin(msgpack_unpacker_t* uk, int raw_type)
             reset_head_byte(uk);
             uk->reading_raw_remaining = 0;
 
-            msgpack_unpacker_stack_t* child_stack = _msgpack_unpacker_new_stack();
-            child_stack->parent = uk->stack;
-            uk->stack = child_stack;
-
+            _msgpack_unpacker_stack_push(uk, STACK_TYPE_RECURSIVE, 1, Qnil);
```

Comment:
Since a child no longer creating a new stack, does this change mean that recursive types will trigger a stack overflow much sooner than before?

Reply:
Possibly, yes. But the stack size if 128, that's quite a deep document, I don't think that's something you are likely to hit..

That said, your comment make me realize there's just no check whatsoever for the stack size :/

Would be worth adding one.

Reply:
Nevermind, there's a check for it:

```c
    if(uk->stack.capacity - uk->stack.depth <= 0) {
        return PRIMITIVE_STACK_TOO_DEEP;
    }
```